effectful/ops/types.py

```diff
@@ -29,28 +30,28 @@ class Operation[**Q, V](abc.ABC):
 
     @abc.abstractmethod
     def __eq__(self, other):
-        raise NotImplementedError
+        raise NotHandled
```

Comment:
I think the errors in `ops.types` should still be `NotImplementedError`s; these are not `Operation`s and `NotImplementedError` is the exception type expected from `abc.abstractmethod`s/`ABC`s.